cmd/common/containerized-gadgets/helpers.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2023 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package containerizedgadgets
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/spf13/viper"
+	"oras.land/oras-go/v2/content/oci"
+	"oras.land/oras-go/v2/registry/remote/auth"
+)
+
+func getLocalOciStore() (*oci.Store, error) {
+	home, err := os.UserHomeDir()
+	if err != nil {
+		return nil, fmt.Errorf("get user home dir: %w", err)
+	}
+
+	return oci.New(home + "/.ig/oci-store")
+}
+
+func getRepositoryFromTag(tag string) string {
+	// For now just assume its something like ghcr.io:443/inspektor-gadget/ig:latest
+	// So we need everything before the last double, which includes also the port
+	splitted := strings.Split(tag, ":")
+	return strings.Join(splitted[:len(splitted)-1], ":")
+}
+
+func getHostString(repoUrlStr string) string {
+	// For now just assume its something like ghcr.io:443/inspektor-gadget/ig
+	// So we need everything before the first slash, which includes also the port
+	return strings.Split(repoUrlStr, "/")[0]
+}
```

Comment:
Should we add some checks to this to be sure it doesn't crash if the output is wrong?

Reply:
I don't think that this will crash. `splitted[:len(splitted)-1]` will return an empty string if there is no `:` or if its an empty string. 
`strings.Split(repoUrlStr, "/")[0]` will also always return at least something (even if it is an empty string)

None the less we should have better warning/error messages regarding this. Should these function return `(string, error)` or print the error themselves?

Reply:
IMO it's always better to have it completely under control and throw an error if the input is not what we expect. Specially in this case that it's coming from the user.